pkg/clusteragent/externalmetrics/provider.go

```diff
@@ -136,14 +135,14 @@ func (p *datadogMetricProvider) getExternalMetric(namespace string, metricSelect
 		parsed = true
 	}
 	if !parsed {
-		return nil, fmt.Errorf("ExternalMetric does not follow DatadogMetric format")
+		return nil, log.Errorf("ExternalMetric does not follow DatadogMetric format: %s", info.Metric)
```

Comment:
I'm wondering it it should be an `log.Errorf` because I don't think the cluster-agent should log an error if the user define a bad configuration. 
Instead the info should be present in the `DatadoMetric.status.conditions`
We can keep a debug log here

Reply:
We cannot update the `DatadogMetric` as we're in the Server part (called by HPA), and this error means we received something that is not in the expected format (so something is wrong in the HPA).

Normally the error should be visible in the HPA `status` as well, but I wanted to highlight as well in the DCA. Maybe it should be a warning, WDYT?

Reply:
understood
can we generate an k8s event instead attached to the DatadogMetric, or HPA/WPA?

Reply:
Not really because the object may just not exist (`DatadogMetric`) and we don't know for which HPA/WPA the query is for at that stage.